src/test/ui/trait-method-number-parameters.stderr

```diff
@@ -0,0 +1,13 @@
+error[E0050]: method `foo` has 2 parameters but the declaration in trait `Foo::foo` has 3
+  --> $DIR/trait-method-number-parameters.rs:17:9
+   |
+LL |       fn foo(&mut self, x: i32, y: i32) -> i32;
+   |              ------------------------- trait requires 3 parameters
+...
+LL | /         &mut self, //~ ERROR
+LL | |         x: i32,
+   | |______________^ expected 3 parameters, found 2
```

Comment:
I think this could be tweaked to actually include either the fn signature or the entire block in order to provide enough context when generating this error in the terminal, but that would be a good follow up tweak.